### PR TITLE
#187972067-should be able to logout

### PR DESCRIPTION
**Pull request by GarrixA**
# This PR is here for:
- making sure a user exits the program, in that moment user will no longer accessing protected routes

### How to test this locally?
- clone the repo by running this in terminal `git clone https://github.com/atlp-rwanda/hackers-ec-Fe.git`
- Open it in your favorite text editor
- In your terminal run `git checkout -b feat-logout-187972067`
- Again run `git pull origin feat-logout-187972067`
- Then run `npm i`
- Then `npm run dev`
- Open app in browser 
- login a user. either admin, seller or buyer
- on each profile click on logout 
- check in your data base the invalidated tokens 
- check if added token can be able to access the protected routes 

## Pivotal tracker Id

       187972067

### CODE REVIEW

**Review by YvetteNyibuka — APPROVED** [review]

File: src/components/DashboardSideNav.tsx
Comment (kwizera-bonheur25) [inline src/components/DashboardSideNav.tsx:1]: Screenshot 2024-07-21 at 19 09 46

Try to change this fixed image to be dynamic for logged user

**Review by kwizera-bonheur25 — CHANGES_REQUESTED** [review]

**Review by kwizera-bonheur25 — APPROVED** [review]

File: src/components/ProfileDropdown.tsx [hunk 1]
```diff
@@ -14,6 +19,25 @@ const ProfileDropdown = ({ image }: ProfileDropdownProps) => {
...
 
+	const dispatch = useAppDispatch();
+	const { isLoading } = useAppSelector((state) => state.logout);
+	const { showErrorMessage, showWorningMessage } = useToast();
+
+	const handleLogout = () => {
+		try {
+			dispatch(Logout());
+			showWorningMessage("You've logged out!");
+		} catch (error) {
```
Comment (leandreAlly) [inline src/components/ProfileDropdown.tsx:31]: let's display messages from backend not hardcoded one, same case on error, always unwrap your dispatch to  avoid unnecessary issues.

File: src/components/cards/LogoutCard.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+const LogoutCard = () => {
+	const dispatch = useAppDispatch();
+	const { isLoading } = useAppSelector((state) => state.logout);
+	const { showErrorMessage, showWorningMessage } = useToast();
+
+	const handleLogout = () => {
+		try {
+			dispatch(Logout());
+			showWorningMessage("You've logged out!");
+		} catch (error) {
```
Comment (leandreAlly) [inline src/components/cards/LogoutCard.tsx:18]: same case here, let's get message from backend response not hardcoded one

**Review by leandreAlly — CHANGES_REQUESTED** [review]

File: src/components/ProfileDropdown.tsx (hunk 1 above)
Comment (GarrixA) [reply src/components/ProfileDropdown.tsx:31]: Done

File: src/components/cards/LogoutCard.tsx (hunk 2 above)
Comment (GarrixA) [reply src/components/cards/LogoutCard.tsx:18]: Done

**Review by leandreAlly — APPROVED** [review]

**Pull request merged**